scripts/build_example

```diff
@@ -4,6 +4,12 @@ readonly SCRIPTS_DIR="$(dirname "$0")"
 # shellcheck source=scripts/common
 source "${SCRIPTS_DIR}/common"
 
+# Get default target architecture for Rust
+llvm_target="$(rustc -Z unstable-options --print target-spec-json | grep llvm-target)"
+IFS=': ' # set delimiter
+read -ra str_split <<< "$llvm_target"
+readonly ARCH="${str_split[1]:1:-2}"
```

Comment:
This seems to be a fair amount of logic that is not immediately readable (to me at least).

If it's not necessary, I suggest we leave it out for the time being. I think the people at Arm trying to compile for a different architecture are better served by the Rust `runner` at this point and going forward. WDYT?

Reply:
Yeah. I don't like it very much either. I'll close this PR for now.

Reply:
Although, looks like we need this fix to be able to run the commands on macOS.

Reply:
Maybe put something in `scripts/common` that respects any `RUST_HOST_TARGET` that is already set in the environment, but if not set picks a default [according to](https://github.com/project-oak/oak/blob/main/scripts/common#L78-L87) `$OSTYPE` ?